doc/feature/spikeReport.md

```diff
@@ -0,0 +1,327 @@
+brion::SpikeReport, but it must be possible to do so by a higher level class
+based on it.
+
+* Provide a container for spikes which is sorted by timestamp, storage
+  efficient, and random accessible at the <sup>n</sup>th position in amortized
+  constant time (provided the data has been already fetched). This is basically
+  a std::vector, but only read access is needed.
+
+* Allow partial data loading for file access.
+
+* Allow the selection of a subset of the cells present in the data source to be
+  reported. In stream-based reports this information should be made available
+  to the sender to allow filtering at the source. In the ZeroEQ implementation,
+  the same channel needed for communicating the desired target could be used
+  for the handshaking required to ensure that the first spikes are not missed.
+
+* As few details as possible that are specific to streams should be exposed.
+  This means that it should be easy to write client code that works the same
+  for both file based and stream based reports without special conditions for
+  stream based reports.
+
+* For stream reports the user will be able to:
+ * Retrieve all the data until the end of the stream in a simple loop without
+   doing any active wait (blocking calls are allowed).
+ * Know if the end of stream has been reached.
+ * Get all the data received so far without locking.
+ * Get information about the latest timestamp reached by the source in the
+   sender side even when there are no spikes at all (currently if the spike
+   report is completely empty it's not possible to make any progress until the
+   producer reaches the end of the report).
+
+* When data for a time interval is requested, the time intervals are always
+  closed on the left and open on the right.
+
+## General remarks
+
+It has been agreed that unifying both the file and the stream cases under
+a stream-like interface (in the sense of std::iostream or POSIX file descriptor
+IO) seems to be the best abstraction at the lower level. Providing random
+access and out-of-core support is complicated at the lower level because
+internal buffering is needed and this can only be implementated efficiently if
+some assumptions about how client code will use the API are made. The most
+important characteristic of the current proposal is that all operations can only
+move forward, so each operation starts where the previous one left over.
+This way there is no need to make any distinction between the file and stream
+based cases. Random access is partially supported by a function that does
+forward skip.
+
+The current proposal defers important design decisions to the higher level API,
+but the its benefit is that it provides a relatively simply API for the most
+critical part.
+
+## API
+
+Thread-safey is only guaranteed for const access.
+
+### Helper types
+
+    struct Spike
+    {
+        Spike( float timestamp_, uint32_t gid_ );
+        float timestamp;
```

Comment:
No particular reason for using timestamp instead of time.

I don't think precision is going to be a problem in the general case, but I think it's a problem for reproducibility if decimal representations are used somewhere in the pipeline (e.g. 0.1 does not have an exact floating point equivalent, no matter how much precision is used). However, the problem is not solved by using double instead of float.